vm/vm.go

```diff
@@ -258,6 +265,21 @@ func (v *vm) Execute(txns []core.Transaction, declaredClasses []core.Class, bloc
 	return context.actualFees, traces, nil
 }
 
+func feltToCChar(value *felt.Felt) *C.char {
+	if value == nil {
+		value = new(felt.Felt).SetUint64(1)
+	}
+	bytes := value.Bytes()
+	return (*C.char)(unsafe.Pointer(&bytes[0]))
```

Comment:
this won't work, `bytes`will go out of scope as this function returns, and the pointer you are returning will point to garbage.